### PR TITLE
build: run four nodes in acceptance tests

### PR DESCRIPTION
**Pull request by tbg**
This is required for

https://github.com/cockroachdb/cockroach/pull/17272

and merits a separate review.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -268,8 +269,9 @@ upload-coverage: $(BOOTSTRAP_TARGET)
 	@build/upload-coverage.sh
 
 .PHONY: acceptance
+acceptance: TESTTIMEOUT := $(ACCEPTANCETIMEOUT)
```
Comment (benesch) [inline Makefile]: `export TESTTIMEOUT := $(ACCEPTANCETIMEOUT)` should do the same without the explicit `env` below!

**Review by benesch — APPROVED** [review]

File: pkg/acceptance/run.sh [hunk 2]
```diff
@@ -10,4 +10,4 @@ set -euxo pipefail
 mkdir -p artifacts/acceptance
 export TMPDIR=$PWD/artifacts/acceptance
 
-make test PKG=./pkg/acceptance TAGS=acceptance TESTFLAGS="${TESTFLAGS--v -nodes 3} -l $TMPDIR"
+make test PKG=./pkg/acceptance TESTTIMEOUT="${TESTTIMEOUT-30m}" TAGS=acceptance TESTFLAGS="${TESTFLAGS--v -nodes 4} -l $TMPDIR"
```
Comment (a-robinson) [inline pkg/acceptance/run.sh]: Shouldn't this be `${TESTTIMEOUT:-30m}` (with the `:`)?

**Review by a-robinson — APPROVED** [review]

File: Makefile (hunk 1 above)
Comment (tschottdorf) [inline Makefile]: Done.

File: pkg/acceptance/run.sh (hunk 2 above)
Comment (tschottdorf) [inline pkg/acceptance/run.sh]: If you care about the case in which the variable is set but null? Anecdotally we're using `-` over `:-` in most places.

**Pull request merged**